lib/ansible/modules/cloud/azure/azure_rm_recoveryservicesvault.py

```diff
@@ -0,0 +1,447 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Zim Kalinowski, (@zikalino)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_recoveryservicesvault
+version_added: '2.9'
+short_description: Manage Azure Vault instance.
+description:
+  - 'Create, update and delete instance of Azure Vault.'
+options:
+  resource_group:
+    description:
+      - >-
+        The name of the resource group where the recovery services vault is
+        present.
+    required: true
+    type: str
+  name:
+    description:
+      - Resource name associated with the resource.
+    required: true
+    type: str
+  e_tag:
+    description:
+      - Optional ETag.
+    type: str
+  location:
+    description:
+      - Resource location.
+    type: str
+  upgrade_details:
```

Comment:
need to check if upgrade_details should be here. I think it may be readonly. is there anything here that we can actually update?